pkg/config/cmd/cmd.go

```diff
@@ -57,47 +108,112 @@ func HaltOnError(fn AfterFunc) AfterFunc {
 	}
 }
 
-func encodeAndCreate(info *resource.Info, namespace string, obj runtime.Object) (runtime.Object, error) {
+// Create is the default create operation for a generic resource.
+func Create(info *resource.Info, namespace string, obj runtime.Object) (runtime.Object, error) {
 	return resource.NewHelper(info.Client, info.Mapping).Create(namespace, false, obj)
 }
 
-// Create attempts to create each item generically, gathering all errors in the
-// event a failure occurs. The contents of list will be updated to include the
-// version from the server.
-func (b *Bulk) Create(list *kapi.List, namespace string) []error {
-	resourceMapper := &resource.Mapper{ObjectTyper: b.Typer, RESTMapper: b.Mapper, ClientMapper: resource.ClientMapperFunc(b.RESTClientFactory)}
-	after := b.After
-	if after == nil {
-		after = func(*resource.Info, error) bool { return false }
+func NoOp(info *resource.Info, namespace string, obj runtime.Object) (runtime.Object, error) {
+	return info.Object, nil
+}
+
+func labelSuffix(set map[string]string) string {
+	if len(set) == 0 {
+		return ""
+	}
+	return fmt.Sprintf(" with label %s", labels.SelectorFromSet(set).String())
+}
+
+func CreateMessage(labels map[string]string) string {
+	return fmt.Sprintf("Creating resources%s", labelSuffix(labels))
+}
+
+type BulkAction struct {
+	// required setup
+	Bulk        Bulk
+	Out, ErrOut io.Writer
+
+	// flags
+	Output      string
+	DryRun      bool
+	StopOnError bool
+
+	// output modifiers
+	Action string
+}
+
+// BindForAction sets flags on this action for when setting -o should only change how the operation results are displayed.
+// Passing -o is changing the default output format.
+func (b *BulkAction) BindForAction(flags *pflag.FlagSet) {
+	flags.StringVarP(&b.Output, "output", "o", "", "Output mode. Use \"-o name\" for shorter output (resource/name).")
+	flags.BoolVar(&b.DryRun, "dry-run", false, "If true, show the result of the operation without performing it.")
```

Comment:
Yes.

On Mon, Apr 18, 2016 at 8:30 AM, David Eads <notifications@github.com>
wrote:

> In pkg/config/cmd/cmd.go
> <https://github.com/openshift/origin/pull/8539#discussion_r60049050>:
>
> > +	Out, ErrOut io.Writer
> > +
> > +	// flags
> > +	Output      string
> > +	DryRun      bool
> > +	StopOnError bool
> > +
> > +	// output modifiers
> > +	Action string
> > +}
> > +
> > +// BindForAction sets flags on this action for when setting -o should only change how the operation results are displayed.
> > +// Passing -o is changing the default output format.
> > +func (b *BulkAction) BindForAction(flags *pflag.FlagSet) {
> > +	flags.StringVarP(&b.Output, "output", "o", "", "Output mode. Use \"-o name\" for shorter output (resource/name).")
> > +	flags.BoolVar(&b.DryRun, "dry-run", false, "If true, show the result of the operation without performing it.")
>
> Seems like you'll want variants for dry-run and confirm. This looks like
> the correct pattern for our reconcile commands.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8539/files/e44ca5593b4d8049f941c4573a5a245248f6481a#r60049050>
>